mypy/build.py

```diff
@@ -877,10 +918,10 @@ def find() -> Optional[str]:
         # many elements of lib_path don't even have a subdirectory 'foo/bar'.  Discover
         # that only once and cache it for when we look for modules like 'foo.bar.blah'
         # that will require the same subdirectory.
-        components = id.split('.')
-        dir_chain = os.sep.join(components[:-1])  # e.g., 'foo/bar'
-        if (dir_chain, lib_path) not in find_module_dir_cache:
-            dirs = []
+        # TODO (ethanhs): refactor to use lru_cache on each of these searches
+        dirs = find_module_dir_cache.get((dir_chain, lib_path), [])
+        if not dirs:
```

Comment:
This is not equivalent to the previous code. Your new code ignores the cache if if says that `dirs == []`, which seems undesirable.